src/main/java/com/radixdlt/crypto/SignaturesImpl.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.crypto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.radixdlt.serialization.DsonOutput;
+import com.radixdlt.serialization.SerializerConstants;
+import com.radixdlt.serialization.SerializerDummy;
+
+import java.util.Map;
+
+/**
+ * A set of cryptographically secure signatures, specifying which signature was used
+ */
+public class SignaturesImpl<T extends Signature> implements Signatures {
+    // Placeholder for the serializer ID
+    @JsonProperty(SerializerConstants.SERIALIZER_NAME)
+    @DsonOutput(DsonOutput.Output.ALL)
+    private SerializerDummy serializer = SerializerDummy.DUMMY;
+
+    @JsonProperty("version")
+    @DsonOutput(DsonOutput.Output.ALL)
+    private short version = 100;
+
+    @JsonProperty("signatures")
+    @DsonOutput(DsonOutput.Output.ALL)
+    private ImmutableMap<ECPublicKey, T> keyToSignature;
```

Comment:
this does not work, need to use same kind of solution used by `permissions` in `MutableSupplyTokenDefinitionParticle`

Reply:
We probably should have a specific de/serializer for ECPublicKey.

Reply:
The question remains how to implement the `setter`, going from `String` to `Signature`, I have not figured out that bit yet....